### PR TITLE
[PyPI] Add option to skip uploading documentation

### PR DESCRIPTION
**Pull request by lukeyeager**
*Fix https://github.com/travis-ci/dpl/issues/334, close https://github.com/travis-ci/dpl/pull/349*

This is a rebase and slight refactor of work from @eric-takealot in https://github.com/travis-ci/dpl/pull/349 after a merge conflict from https://github.com/travis-ci/dpl/pull/396.

> Has this been tested?
@BanzaiMan https://github.com/travis-ci/dpl/pull/349#issuecomment-149536356

With `skip_upload_docs: true`: https://travis-ci.org/lukeyeager/DIGITS/jobs/172784593
With `skip_upload_docs: false`: https://travis-ci.org/lukeyeager/DIGITS/jobs/172785158

### CODE REVIEW

File: lib/dpl/provider/pypi.rb [hunk 1]
```diff
@@ -96,7 +96,10 @@ def push_app
         context.shell "python setup.py #{pypi_distributions}"
         context.shell "twine upload -r pypi dist/*"
         context.shell "rm -rf dist/*"
-        context.shell "python setup.py upload_docs #{pypi_docs_dir_option} -r #{pypi_server}"
+        unless options[:skip_upload_docs]
+          log "Uploading documentation (skip with \"skip_upload_docs: true\")"
+          context.shell "python setup.py upload_docs #{pypi_docs_dir_option} -r #{pypi_server}"
```
Comment (lukeyeager) [inline lib/dpl/provider/pypi.rb]: Yes - you can run it multiple times. That was the behavior before this PR anyway, and I believe I've proved that it works as expected with my tests linked in the OP.

**Comment by lukeyeager** [comment]
Is there anything else you need from me?

**Comment by lukeyeager** [comment]
Rebased

**Comment by nicoddemus** [comment]
Thanks for merging this.

When can we be expected to start using this in production?

**Comment by BanzaiMan** [comment]
Not clear at the moment. The `master` branch is failing, so no release can be planned until that's fixed. Sorry.

**Comment by nicoddemus** [comment]
> Not clear at the moment. The master branch is failing, so no release can be planned until that's fixed. Sorry.

No problem at all. Is there an issue or something that I can subscribe to in order to know when this feature is available in production?

**Pull request merged**